sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -380,8 +380,8 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
         DataSourceV2Relation.create(t)
     }
 
-    val command = modeForDSV2 match {
-      case SaveMode.Append =>
+    val command = mode match {
+      case SaveMode.Append | SaveMode.ErrorIfExists | SaveMode.Ignore =>
```

Comment:
A note to future readers: this is the previous behavior, that non-overwrite mode means append. This is due to the bad design of `DataFrameWriter`: we only need to know overwrite or not when calling `insert`, but `DataFrameWriter` gives you a save mode. Since the default save mode is `ErrorIfExists`, treating non-overwrite mode as append is a reasonable compromise.

Note that, we don't have this problem in the new `DataFrameWriterV2`.

Reply:
Looks like the previous version used this:

```
      InsertIntoTable(
        table = UnresolvedRelation(tableIdent),
        partition = Map.empty[String, Option[String]],
        query = df.logicalPlan,
        overwrite = mode == SaveMode.Overwrite, // << Either overwrite or append
        ifPartitionNotExists = false)
```

So I agree that this is using the same behavior that v1 did.